sdk/src/Services/DynamoDBv2/Custom/DocumentModel/DocumentTransactWrite.cs

```diff
@@ -325,6 +362,38 @@ internal void AddDocumentToUpdateHelper(Document document, Key key, TransactWrit
             });
         }
 
+        internal void AddKeyToUpdateHelper(Key key, Expression updateExpression, TransactWriteItemOperationConfig operationConfig)
+        {
+            if (operationConfig?.ConditionalExpression != null)
+            {
+                // Filter attribute names and values from the update expression that are also found in the condition expression
+                // If the value is different for the same key between the two expressions, an exception is thrown
+                // This ensures no exception is thrown when the attribute names and values of the update expression are merged into the request object
+                if (!TryFilterDuplicates(updateExpression.ExpressionAttributeValues,
+                        operationConfig.ConditionalExpression.ExpressionAttributeValues, out var attributeValues) ||
+                    !TryFilterDuplicates(updateExpression.ExpressionAttributeNames,
+                        operationConfig.ConditionalExpression.ExpressionAttributeNames, out var attributeNames))
+                {
+                    throw new ArgumentException("Conflicting values for key", "updateExpression");
```

Comment:
1. Could you expand this exception message to explain how to correct this, and which key triggered it if not too much more work?
2. Could you also cover the duplicate filtering via tests - assert that we're seeing this `ArgumentException` when expected?

Reply:
Done, please check if it's okay this way